### PR TITLE
Add linkedEvent column to OrderPickupEvent Table

### PR DESCRIPTION
**Pull request by nik-dange**
# Description

Need a way to tie order pickup events to real events so that frontend can display things like the event graphic

## Changes

- Added a column to OrderPickupEvents table to link to Events
- Added an accompanying relation

# Type of Change

- [ ] Patch (non-breaking change/bugfix)
- [x] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to not work as
      expected)
- [ ] Documentation (A change to a README/description)
- [ ] Continuous Integration/DevOps Change (Related to deployment steps, continuous integration
      workflows, linting, etc.)
- [ ] Other: (Fill In) 

If you've selected Patch, Minor, or Major as your change type, **make sure to bump the version before merging in `package.json`!**
# Testing

I have tested that my changes fully resolve the linked issue ...

- [ ] locally.
- [ ] on the testing API/testing database.
- [ ] with appropriate Postman routes. Screenshots are included below.

# Checklist

- [ ] I have performed a self-review of my own code.
- [ ] I have followed the style guidelines of this project.
- [ ] I have appropriately edited the API version in the `package.json` file.
- [ ] My changes produce no new warnings.

# Screenshots

Please include a screenshot of your Postman testing passing successfully.

### CODE REVIEW

**Comment by nik-dange** [comment]
I do api change tmrw I am sleepy

File: types/ApiRequests.ts [hunk 1]
```diff
@@ -308,6 +309,7 @@ export interface OrderPickupEvent {
   end: Date;
   description: string;
   orderLimit: number;
+  linkedEvent?: Event;
```
Comment (nik-dange) [inline types/ApiRequests.ts:312]: I made this of type `Event`, should it instead be of type `UUID`?

File: api/validators/MerchStoreRequests.ts [hunk 2]
```diff
@@ -258,6 +262,10 @@ export class OrderPickupEvent implements IOrderPickupEvent {
   @IsDefined()
   @Min(1)
   orderLimit: number;
+
+  @IsOptional()
+  @IsUUID()
+  linkedEvent?: EventModel;
```
Comment (nik-dange) [inline api/validators/MerchStoreRequests.ts:268]: Validating this with type EventModel doesn't seem correct to me although it satisfies any type errors, is this validation even necessary for this optional field?

File: types/ApiRequests.ts (hunk 1 above)
Comment (dowhep) [reply types/ApiRequests.ts:312]: I think event is fine for the api response, saving us an extra api request

File: api/validators/MerchStoreRequests.ts (hunk 2 above)
Comment (dowhep) [reply api/validators/MerchStoreRequests.ts:268]: yeah i suppose just for the possibility of storing non-uuid stuff in the database

File: tests/data/MerchFactory.ts [hunk 3]
```diff
@@ -110,6 +110,7 @@ export class MerchFactory {
       orderLimit: FactoryUtils.getRandomNumber(1, 5),
       status: OrderPickupEventStatus.ACTIVE,
       orders: [],
+      linkedEvent: null,
```
Comment (nik-dange) [inline tests/data/MerchFactory.ts:113]: Should I import the EventFactory and generate an actual dummy event to put in here? Are there any other test cases that could be beneficial for this feature?

File: models/OrderPickupEventModel.ts [hunk 4]
```diff
@@ -29,6 +30,10 @@ export class OrderPickupEventModel extends BaseEntity {
   @JoinColumn({ name: 'order' })
   orders: OrderModel[];
 
+  @OneToOne((type) => EventModel, { nullable: true })
+  @JoinColumn({ name: 'linkedEvent' })
+  linkedEvent: EventModel;
+
```
Comment (nik-dange) [inline models/OrderPickupEventModel.ts:36]: Can you double check this relation–I found a few different syntax examples online but I believe this establishes the correct one-way relation.

File: tests/data/MerchFactory.ts (hunk 3 above)
Comment (dowhep) [reply tests/data/MerchFactory.ts:113]: I don't actually think so. I actually don't think it is clean to attach a dummy event everytime we mock a pickupEvent as sometimes pickupEvent doesn't have an actual linkedEvent attached.

I don't believe it is necessary to implement a solution for this now, but for future reference, if there is a new feature that actually depends on linkedEvent, we should come up with a nice way (such as taking an optional boolean flag variable to attach event) to attach dummy linkedEvent to the mocked pickupEvent.

File: models/OrderPickupEventModel.ts (hunk 4 above)
Comment (dowhep) [reply models/OrderPickupEventModel.ts:36]: This looks good to me.

**Review by dowhep — CHANGES_REQUESTED** [review]
A few nits. Remember to attach postman screenshots.

File: services/MerchStoreService.ts [hunk 5]
```diff
@@ -1033,23 +1034,38 @@ export default class MerchStoreService {
...
 
   public async editPickupEvent(uuid: Uuid, changes: OrderPickupEventEdit): Promise<OrderPickupEventModel> {
     return this.transactions.readWrite(async (txn) => {
       const orderPickupEventRepository = Repositories.merchOrderPickupEvent(txn);
       const pickupEvent = await orderPickupEventRepository.findByUuid(uuid);
       const updatedPickupEvent = OrderPickupEventModel.merge(pickupEvent, changes);
+
+      if (changes.linkedEventUuid) {
+        const linkedRegularEvent = await this.getLinkedRegularEvent(changes.linkedEventUuid);
+        updatedPickupEvent.linkedEvent = linkedRegularEvent;
```
[the remark this answers is not in the document]
Comment (nik-dange) [reply services/MerchStoreService.ts:1066]: This method is called when a pickup event is edited, so only the UUID of the linkedEvent would be changed. I made a call to the upsert function on line 1076, which should be the only call necessary to save this change, right?

**Comment by nik-dange** [comment]
Screenshot 2024-01-15 at 3 28 25 PM
Screenshot 2024-01-15 at 3 20 04 PM
Screenshot 2024-01-15 at 3 06 00 PM

**Review by dowhep — APPROVED** [review]
LGTM 🚀 🚀 🚀 🚀 🚀

**Pull request merged**